### PR TITLE
ISPN-14914 Admin view for cache templates

### PR DESCRIPTION
**Pull request by dpanshug**
https://issues.redhat.com/browse/ISPN-14914

### CODE REVIEW

**Comment by dpanshug** [comment]
To be merged after #368

**Comment by andyuk1986** [comment]
I have rebased, and fixed the test. Looks good to me.

**Comment by dpanshug** [comment]
Thanks @andyuk1986, I have fixed that problem.

File: src/app/services/cachesHook.ts [hunk 1]
```diff
@@ -113,3 +113,29 @@ export function useSetAvailableCache(cacheName: string) {
...
 }
+
+export function useFetchCacheTemplates(cmName) {
+  const [cacheTemplates, setCacheTemplates] = useState<CacheConfig[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    if (loading) {
+      console.log('d');
```
Comment (karesti) [inline src/app/services/cachesHook.ts:124]: remove

File: src/app/services/cachesHook.ts [hunk 2]
```diff
@@ -113,3 +113,29 @@ export function useSetAvailableCache(cacheName: string) {
...
+      ConsoleServices.dataContainer()
+        .getCacheConfigurationTemplates(cmName)
+        .then((eitherConfigs) => {
+          setLoading(false);
+          if (eitherConfigs.isRight()) {
+            console.log('eitherConfigs', eitherConfigs);
```
Comment (karesti) [inline src/app/services/cachesHook.ts:130]: remove

File: src/app/services/cachesHook.ts [hunk 3]
```diff
@@ -113,3 +113,29 @@ export function useSetAvailableCache(cacheName: string) {
...
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    if (loading) {
+      console.log('d');
+      ConsoleServices.dataContainer()
+        .getCacheConfigurationTemplates(cmName)
+        .then((eitherConfigs) => {
+          setLoading(false);
```
Comment (karesti) [inline src/app/services/cachesHook.ts:128]: if we do finally setLoading false no need to do it here, right ?

**Review by karesti — CHANGES_REQUESTED** [review]
some console logs forgotten 
other wise

**Comment by karesti** [comment]
@dpanshug related failing tests

**Review by karesti — APPROVED** [review]

**Pull request merged**